File: test/data/typescript/test05/StateState4.ts
import {BaseState} from "./BaseState"
import {IControllee} from "./IControllee"
import {StateController} from "./StateController"
export class StateState4 extends BaseState
{
    private _stateController: StateController;
    private _controlleeImp: IControllee;
    public constructor(stateController: StateController, controllee: IControllee)
    {
        super(controllee);
        this._stateController = stateController;
        this._controlleeImp = controllee;
    }
    public  GetStateName(): string
    {
        return "State4";
    }
    public GetParent(): BaseState | null
    {
        return null;
    }
    public GetStateID(): number
    {
        return 4;
    }
}

File: test/data/cpp/test05/StateState4.cpp
#include "StateState4.h"
StateState4::StateState4(StateController* stateController, IControllee* controllee) : BaseState(controllee)
{
    _stateController = stateController;
    _controllee = controllee;
}
StateState4::~StateState4()
{
}
const char* StateState4::GetStateName()
{
    return "State4";
}
BaseState* StateState4::GetParent()
{
    return nullptr;
}
int StateState4::GetStateID()
{
    return 4;
}